src/omni_epd/displays/waveshare_display.py

```diff
@@ -125,25 +159,106 @@ def __init__(self, deviceName, config):
     def get_supported_devices():
         result = []
 
-        deviceList = ["epd1in54b", "epd1in54b_V2", "epd1in54c",
-                      "epd2in13b_V3", "epd2in13bc", "epd2in66b",
-                      "epd2in7b", "epd2in7b_V2", "epd2in9b_V3",
-                      "epd2in9bc", "epd4in2b_V2", "epd4in2bc",
-                      "epd5in83b_V2", "epd5in83bc", "epd7in5b_HD",
-                      "epd7in5b_V2", "epd7in5bc"]
-
         # python libs for this might not be installed - that's ok, return nothing
         if(WaveshareTriColorDisplay.check_module_installed('waveshare_epd')):
-            result = [f"{WaveshareTriColorDisplay.pkg_name}.{n}" for n in deviceList]
+            result = [f"{WaveshareTriColorDisplay.pkg_name}.{n}" for n in WaveshareTriColorDisplay.deviceMap]
 
         return result
 
     def prepare(self):
         self._device.init()
 
     def _display(self, image):
-        # send the black/white image and blank second image (safer since some require data)
-        self._device.display(self._device.getbuffer(image), [0x00] * (int(self.width/8) * self.height))
+
+        if(self.mode == 'bw'):
+            # send the black/white image and blank second image (safer since some drivers require data)
+            self._device.display(self._device.getbuffer(image), [0x00] * (int(self.width/8) * self.height))
+        else:
+            # apply the color filter to get a 3 color image
+            image = self._filterImage(image)
+
+            # separate out black from the other color
+            img_black = image.copy()
+            img_black.putpalette((0, 0, 0, 255, 255, 255, 255, 255, 255) + (0, 0, 0)*253)
+
+            img_color = image.copy()
+            img_color.putpalette((255, 255, 255, 0, 0, 0, 255, 255, 255) + (0, 0, 0)*253)
```

Comment:
```suggestion
            img_black.putpalette((255, 255, 255, 0, 0, 0, 255, 255, 255) + (0, 0, 0)*253)

            img_color = image.copy()
            img_color.putpalette((255, 255, 255, 255, 255, 255, 0, 0, 0) + (0, 0, 0)*253)
```
These are in the wrong order.